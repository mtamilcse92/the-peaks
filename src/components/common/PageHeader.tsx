import React, { ReactNode } from "react";
import styles from "./PageHeader.module.scss";

export type Props = {
    title: string
    rhsElement?: ReactNode;
};

const PageHeader: React.FC<Props> = ({ title, rhsElement }) => {
  
  return (
    <header className={styles.header}>
        <span className={styles.heading}>{title}</span>
        {rhsElement && <div className={styles.rhs}>
          {rhsElement}
        </div>}
      </header>
  );
};

export default PageHeader;
