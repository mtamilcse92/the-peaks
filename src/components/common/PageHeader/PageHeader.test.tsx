import React from "react";
import { render } from "@testing-library/react";
import PageHeader from "./index";

describe("<PageHeader  /> spec", () => {
  it("renders the PageHeader", () => {
    const WithOutRHS = render(<PageHeader title="All Bookmarks" />);
    const WithCustomRHS = render(
      <PageHeader
        title="All Bookmarks"
        rhsElement={<div>Show All Bookmarks</div>}
      />
    );
    expect(WithCustomRHS.getByText("Show All Bookmarks")).toBeInTheDocument();
    expect(WithCustomRHS).toMatchSnapshot();
    expect(WithOutRHS).toMatchSnapshot();
  });
});
