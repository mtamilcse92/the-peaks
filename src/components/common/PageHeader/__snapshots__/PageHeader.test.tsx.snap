// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader  /> spec renders the PageHeader 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <span
          class="heading"
        >
          All Bookmarks
        </span>
      </header>
    </div>
    <div>
      <header
        class="header"
      >
        <span
          class="heading"
        >
          All Bookmarks
        </span>
        <div
          class="rhs"
        >
          <div>
            Show All Bookmarks
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <span
        class="heading"
      >
        All Bookmarks
      </span>
      <div
        class="rhs"
      >
        <div>
          Show All Bookmarks
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<PageHeader  /> spec renders the PageHeader 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <span
          class="heading"
        >
          All Bookmarks
        </span>
      </header>
    </div>
    <div>
      <header
        class="header"
      >
        <span
          class="heading"
        >
          All Bookmarks
        </span>
        <div
          class="rhs"
        >
          <div>
            Show All Bookmarks
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <span
        class="heading"
      >
        All Bookmarks
      </span>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
